docs/src/content/docs/basic-concepts.mdx

```diff
@@ -0,0 +1,116 @@
+---
+title: Basic Concepts
+description: Understanding the core concepts of GenSX
+---
+
+GenSX is a workflow engine designed for building complex LLM applications. It's built around functional, reusable components that are composed using JSX to create and orchestrate complicated workflows.
+
+## Components
+
+Components are the building blocks of GenSX applications; they're pure TypeScript functions that:
+
+- Accept props as input
+- Return a single output or another component
+- Have zero side effects
```

Comment:
Not true, you can do whatever you want in a component, write to S3, update a db row, etc.

Reply:
good point

Reply:
removed this bullet